### PR TITLE
Adding unit tests for the isc-dhcp-lease-poller

### PR DESCRIPTION
**Pull request by tldavies**
These test are from https://github.com/RackHD/on-dhcp-proxy/pull/39/files that was closed.

### CODE REVIEW

**Comment by tldavies** [comment]
@benbp @jlongever @RackHD/corecommitters

**Comment by yyscamper** [comment]
:+1:

**Comment by VulpesArtificem** [comment]
👍

File: spec/lib/jobs/isc-dhcp-poller-job-spec.js [hunk 1]
```diff
@@ -127,6 +135,75 @@ describe('ISC DHCP Poller Job', function () {
...
+        it('should stop the queue even when there is no tail', function (){
+            var DHCPPoller = new this.Jobclass({}, {}, uuid.v4());
+            var spy = sinon.spy(PromiseQueue.prototype, 'stop');
+
+            DHCPPoller._cleanup();
+
+            expect(spy).to.have.been.calledOnce;
+        });
+
+        it("should unwatch and remove the tail");
```
Comment (DavidjohnBlodgett) [inline spec/lib/jobs/isc-dhcp-poller-job-spec.js]: is this a test (it statement) that wants to be removed?

Comment (tldavies) [inline spec/lib/jobs/isc-dhcp-poller-job-spec.js]: it is a pending test. Like we should a test to test it. But do not at the current time.

**Pull request merged**